src/storage/leveldb/utxo_index.ts

```diff
@@ -0,0 +1,223 @@
+      let maxkey = `${authorities}:${token}:`;
+
+      if (options.amount_bigger_than) {
+        const minvalbuf = Buffer.alloc(8);
+        minvalbuf.writeBigUint64BE(BigInt(options.amount_bigger_than));
+        minkey = `${minkey}${minvalbuf.toString('hex')}`;
+      }
+      if (options.amount_smaller_than !== undefined) {
+        const maxvalbuf = Buffer.alloc(8);
+        maxvalbuf.writeBigUint64BE(BigInt(options.amount_smaller_than + 1));
+        maxkey = `${maxkey}${options.filter_address}:${maxvalbuf.toString('hex')}`;
+      } else {
+        const lastChar = String.fromCharCode(options.filter_address.charCodeAt(options.filter_address.length - 1) + 1);
+        const maxaddr = `${options.filter_address.slice(0, -1)}${lastChar}`;
+        maxkey = `${maxkey}${maxaddr}:`;
+      }
+
+      itOptions.gte = minkey;
+      itOptions.lte = maxkey;
+    } else {
+      // No need to filter by address, just tokens
+      db = this.tokenUtxoDB;
+      let minkey = `${authorities}:${token}:`;
+      let maxkey = `${authorities}:`;
+
+      if (options.amount_bigger_than) {
+        const minvalbuf = Buffer.alloc(8);
+        minvalbuf.writeBigUint64BE(BigInt(options.amount_bigger_than));
+        minkey = `${minkey}${minvalbuf.toString('hex')}`;
+      }
+      if (options.amount_smaller_than !== undefined) {
+        const maxvalbuf = Buffer.alloc(8);
+        maxvalbuf.writeBigUint64BE(BigInt(options.amount_smaller_than + 1));
+        maxkey = `${maxkey}${token}:${maxvalbuf.toString('hex')}`;
+      } else {
+        const lastChar = String.fromCharCode(token.charCodeAt(token.length - 1) + 1);
+        const maxtoken = `${token.slice(0, -1)}${lastChar}`;
+        maxkey = `${maxkey}${maxtoken}:`;
+      }
+
+      itOptions.gte = minkey;
+      itOptions.lte = maxkey;
+    }
+
+    let sumAmount = 0;
+    let utxoNum = 0;
+    for await (const utxo of db.values(itOptions)) {
+      if (isHeightLocked(utxo) || (options.filter_method && !options.filter_method(utxo))) {
+        continue;
+      }
+      if (options.max_amount && ((sumAmount + utxo.value) > options.max_amount)) {
+        continue;
+      }
+
+      yield utxo;
+
+      utxoNum += 1;
+      sumAmount += utxo.value;
+
+      if ((options.target_amount && sumAmount >= options.target_amount) || (utxoNum >= maxUtxos)) {
+        // We have reached either the target amount or the max number of utxos requested
+        return;
+      }
+    }
+  }
+
+  async saveUtxo(utxo: IUtxo): Promise<void> {
+    await this.utxoDB.put(_utxo_id(utxo), utxo);
+    await this.tokenAddressUtxoDB.put(_token_address_utxo_key(utxo), utxo);
+    await this.tokenUtxoDB.put(_token_utxo_key(utxo), utxo);
+  }
+
+  async clear(): Promise<void> {
+    // This should clear all utxos subdbs
+    await this.utxoDB.db.clear();
+    // await this.utxoDB.clear();
+    // await this.tokenUtxoDB.clear();
+    // await this.tokenAddressUtxoDB.clear();    
```

Comment:
Don't we need this anymore? Why is it commented out? Remove?

Reply:
Removed